aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -403,9 +402,9 @@ cunn_SoftMaxForward(scalar_t *output, scalar_t *input, int classes)
 
   // find the max
   accscalar_t threadMax = ilpReduce<MaxFloat, ILP, scalar_t, accscalar_t>(
-      input, classes, MaxFloat<scalar_t, accscalar_t>(), -THCNumerics<accscalar_t>::max());
+      input, classes, MaxFloat<scalar_t, accscalar_t>(), -std::numeric_limits<accscalar_t>::max());
```

Comment:
ROCm doesn't support calling functions on `std::numeric_limits` in device code, which is why this was still using THCNumerics

Reply:
The functions on numeric_limits aren't marked with `__device__` which is why they don't work with ROCm. CUDA allows them because they're `constexpr`.

(It doesn't matter that the calling functions are themselves device lambdas)